src/uwtools/resources/rocoto.jsonschema

```diff
@@ -0,0 +1,72 @@
+{
+    "definitions": {
+        "dateTimePattern": {
+            "type": "string",
+            "pattern": "^(\\d{12}) (\\d{12}) (\\d{2}):(\\d{2}):(\\d{2}):(\\d{2})$"
+        }
```

Comment:
Circling back on checking if JSON Schema supports defining patterns and reusing them:

It seems that JSONSchema does not support constants/declaring variables. However, I'm finding [discussion](https://github.com/json-schema-org/json-schema-spec/issues/886) online that others want to do a similar thing here, where we can add our regex expression in a constant and reuse the variable when needed. 

It seems to be that you we cannot _directly_ save an expressions to variable and reuse them within the schema. There is no built-in method for defining variables.

The best way to do this was to define `definitions` that holds the reusable section we want to use in multiple places in our schema. In this way we are referencing the the information to avoid duplication of code.

@maddenp-noaa @christinaholtNOAA